### DISCUSSION TITLE
Do I have to create a payment gateway plugin when an external Saleor App handles the payment process?

### DISCUSSION THREAD

**Discussion by nnthuan** [saleor/saleor#10743] [Deployment] [answered]
Do I have to create a payment gateway plugin when an external Saleor App handles the payment process?

https://docs.saleor.io/docs/3.x/developer/checkout#finalizing-checkout-with-saleor-app

I'm looking forward to your response. Thanks!

**Comment by fowczarek** [comment, marked as answer]
When the payment process is handled by Saleor App you don't need to create any plugin.

**Reply by nnthuan** [reply]
Hi @fowczarek, Thank you for your reply. I have one more question:

I don't need to create any plugins. Does this mean that receiving payment events from webhook and order updates are handled by Saleor App?

**Reply by fowczarek** [reply]
Yes, you need to create Saloer App and subscribe to proper payment webhooks. 
More info [here](https://docs.saleor.io/docs/3.x/developer/extending/apps/synchronous-webhooks/payment-webhooks)
